test/functional/test_runner.py

```diff
@@ -218,14 +233,23 @@ def main():
 
     check_script_list(config["environment"]["SRCDIR"])
 
+    if not args.keepcache:
+        shutil.rmtree("%s/test/cache" % config["environment"]["BUILDDIR"], ignore_errors=True)
+
     run_tests(test_list, config["environment"]["SRCDIR"], config["environment"]["BUILDDIR"], config["environment"]["EXEEXT"], args.jobs, args.coverage, passon_args)
 
 def run_tests(test_list, src_dir, build_dir, exeext, jobs=1, enable_coverage=False, args=[]):
-    BOLD = ("","")
-    if os.name == 'posix':
-        # primitive formatting on supported
-        # terminal via ANSI escape sequences:
-        BOLD = ('\033[0m', '\033[1m')
+    # Warn if dashd is already running (unix only)
+    try:
+        if subprocess.check_output(["pidof", "dashd"]) is not None:
```

Comment:
This check always fails for me on mac... `subprocess.check_output(["pidof", "dashd"])` returns `b''` for some reason here.
```
        pidof_output = subprocess.check_output(["pidof", "dashd"])
        if not (pidof_output is None or pidof_output == b''):
```
does the job but maybe there is a better way